runtime/lua/vim/lsp/util.lua

```diff
@@ -2207,6 +2207,50 @@ function M.lookup_section(settings, section)
   return settings
 end
 
+--- Helper function to resolve buffer number.
+--- Returns current buffer if bufnr is not valid
+---@param bufnr integer buffer number
+---@return integer buffer number
+function M.resolve_bufnr(bufnr)
+  return bufnr > 0 and bufnr or api.nvim_get_current_buf()
+end
```

Comment:
I wouldn't share this, it is a trivial function and `:h dev-lua` mentions:

> Interface conventions  
> 
> - When accepting a buffer id, etc., 0 means "current buffer", nil means "all
>   buffers".  Likewise for window id, tabpage id, etc.


This function violates that guideline